application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/internal/MethodWithFrameTypeLabelProvider.java

```diff
@@ -106,7 +106,7 @@ public Image getImage(Object key) {
 			} else if (key instanceof IMCFrame) {
 				method = ((IMCFrame) key).getMethod();
 			}
-			if (method != null) {
+			if ((method != null) & (method.getModifier() != null)) {
```

Comment:
Spotbugs:
```
Error:  High: Questionable use of non-short-circuit logic in org.openjdk.jmc.flightrecorder.ui.pages.internal.MethodWithFrameTypeLabelProvider.getImage(Object) [org.openjdk.jmc.flightrecorder.ui.pages.internal.MethodWithFrameTypeLabelProvider] At MethodWithFrameTypeLabelProvider.java:[line 109] NS_NON_SHORT_CIRCUIT
```
NS: Questionable use of non-short-circuit logic (NS_NON_SHORT_CIRCUIT)

This code seems to be using non-short-circuit logic (e.g., & or |) rather than short-circuit logic (&& or ||). Non-short-circuit logic causes both sides of the expression to be evaluated even when the result can be inferred from knowing the left-hand side. This can be less efficient and can result in errors if the left-hand side guards cases when evaluating the right-hand side can generate an error.